kytos/core/interface.py

```diff
@@ -451,28 +387,60 @@ def add_tags(self, tags: list[int], tag_type: str = 'vlan') -> bool:
     def make_tags_available(
         self,
         controller,
-        tags: Union[int, list[int]],
+        tags: Union[int, list[int], list[list[int]]],
         tag_type: str = 'vlan',
         use_lock: bool = True,
-    ) -> bool:
+    ) -> list[list[int]]:
         """Add a specific tag in available_tags.
 
         Args:
             controller: Kytos controller
             tags: List of tags, there should be 2 items in the list
             tag_type: TAG type value
             use_lock: Boolean to whether use a lock or not
+
+        Return:
+            conflict: Return any values that were not added.
         """
         if isinstance(tags, int):
             tags = [tags] * 2
+        if isinstance(tags[0], int) and tags[0] != tags[1]:
```

Comment:
If an empty list is passed this will raise `IndexError`, it'd be great to also validate this particular case. Also an unsupported `tag_type` would result in `KeyError` too. Both are OK and idiomatic, but should we also re-raise these as TAG exception errors instead? In other parts of te core we do also let to raise `KeyError` but since you're standardazing tag errors, it might be great to also catch and re-raise them accordingly or even just use a base tag error exc with the reason. Wdyt?

Reply:
Now, regarding other wrong values of `list[list[int]` like out of order, or redundant or anything else that might break range addition or diff we might want to validate in the future. My biggest concern is that UNI with tag range are exposed to users, so if passed wrong values and result in other issues, although if we guarantees in the HTTP API boundaries that deep down it's always passing validated then we minimize this surface of issues and then we can document this and always recommend that it's validated only at the highest boundaries like `mef_eline` with UNI using ranges and `topology` that's exposing to external users and each NApp reuse `get_tag_ranges` wdyt?

Reply:
I have not tackled this problem yet. Currently `mef_eline` and `topology` are sending supported values by reusing `get_tag_ranges`. The ranges operations from `tag_ranges.py` could detect out of order ranges but I was thinking only in values managed by kytos. Could this be a separate issue though?

Reply:
>  Currently mef_eline and topology are sending supported values by reusing get_tag_ranges

Great. 

>  Could this be a separate issue though?

Yes, it can be in another issue since it won't block this functionality and the epic is still being delivered. 

> The ranges operations from tag_ranges.py could detect out of order ranges but I was thinking only in values managed by kytos.

Could you elaborate on it? I'm not sure if I followed the only values managed by kytos part. Good thing that `get_tag_ranges` is relatively lightweight. Either we find a way to optimally validate this, or just delegate and document that all tag ranges being passed need to be validated upfront (which is already partly the current case as you've mentioned).

Reply:
I wasn't thinking about the user's inputs bypassing the API requests. The range operations `range_intersection`, `range_difference`, and `range_addition` could be slightly modified to detect out-of-order values. These methods returns the result instead of modifying and can error out if inside the `while` loop a range was detected to be out of range by comparing both `[a, b]` integers and comparing them with the previous range.

Reply:
@Alopalao, great idea, seems very ergonomic, we might as well leverage these existing loops. Let's go for it, but let's also make sure that is a validation issue happens, then it also has a guarantee of rolling back the original `available_tags` and/or `tag_ranges`, OK? Feel free to prototype and early explore it.

Reply:
I tested this idea and the returned values is guaranteed to be correct. But some of the methods do not go over the whole list for example:
The difference between `[[1, 3], [5, 8]]` and `[[3, 4], [9, 10]]` will be called like `range_difference([[1, 3], [5, 8]], [[3, 4], [9, 10]])`. The range [9, 10] is out of order but it will not be detected because it does not affect the result. The method only iterated through `[1, 3]`, `[3, 4]` and `[5,8]`. Would this be okay?

Reply:
Aldo, did you mean `[10, 9]` instead of `[9, 10]` for out of order? 

I find it a bit non intuitive the fact it's performing a difference between "a" and "b" and "b" not being iterated entirely, doesn't that leave margin for a potential corner case? Looks like it was assuming that the list was upfront sorted? I'd be more comfortable with this if either it's iterated entirely or we pay the price to sort the list(s).

Reply:
You are right, overlooked that part. So for `use_tags()` and `make_tags_available()`, I added the argument `check_order: bool = True`. If this option is true, `tags` would go through `get_verified_tags()` to verify/correct the list. I am going to make some changes to the other NApps, the verification adds some time complexity to internal processes that I think are safe to bypass the checking of the order.
Commit [6b3caf9372f4872f2e866d95145a10c29ed43d54](https://github.com/kytos-ng/kytos/pull/428/commits/6b3caf9372f4872f2e866d95145a10c29ed43d54)